walker.go

```diff
@@ -192,9 +249,104 @@ func (s *StatInfo) Sys() interface{} {
 	return s.Stat
 }
 
-func matchPrefix(pattern, name string) (bool, bool) {
-	count := strings.Count(name, string(filepath.Separator))
+func matchPrefix(pattern, name string, isDir bool) (bool, bool) {
 	partial := false
+	if strings.Contains(pattern, "**") {
+		// short-circuit for single "**"
+		if pattern == "**" || pattern == "**/" {
+			return true, isDir
```

Comment:
The `isDir` handling here is quite confusing. Ideally, this should be done in a wrapper layer outside `matchPrefix` and let this function work on strings, not files. Seems like the meaning of the "partial" result here has changed.

Reply:
The `isDir` usage here is probably not needed, adding Walk tests like you suggest should prove that out.

I didn't intend to change the meaning of `partial` results, it is just that `partial` is ambiguous in the context of `**` since we don't know how deep the pattern should match.

For example: `**/baz` should return `partial=true` for `foo/bar` if `bar` is a directory (since `bar` might have children that match our pattern).  But if `bar` is a file, then it cannot possibly contain children so we can return a negative non-partial match.